src/core/server/saved_objects/migrationsv2/actions/bulk_overwrite_transformed_documents.ts

```diff
@@ -71,12 +74,19 @@ export const bulkOverwriteTransformedDocuments = ({
     .then((res) => {
       // Filter out version_conflict_engine_exception since these just mean
       // that another instance already updated these documents
-      const errors = (res.body.items ?? []).filter(
-        (item) => item.index?.error?.type !== 'version_conflict_engine_exception'
-      );
+      const errors = (res.body.items ?? [])
+        .filter((item) => item.index?.error)
+        .map((item) => item.index!.error!)
```

Comment:
wow! so many `?`, `!` 😅  I'd write it as 
```js
.map((item) => item.index?.error)
.filter(Boolean)
.filter(({ type }) => type !== 'version_conflict_engine_exception');
```

Reply:
Yea, the problem is that TS is stupid with map/filter. 

```ts
.map((item) => item.index?.error)
.filter(Boolean)
```

Is not sufficient to have the `| undefined` part removed from `error`. The third line complains that `ErrorContainer | undefined` does not have a `type` property, which is why I kinda was forced to filter first then force-cast using `!`